src/funnels/Swapper.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: © 2020 Lev Livnev <lev@liv.nev.org.uk>
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface RolesLike {
+    function canCall(bytes32, address, address, bytes4) external view returns (bool);
+}
+
+interface GemLike {
+    function balanceOf(address) external view returns (uint256);
+    function transferFrom(address, address, uint256) external;
+}
+
+interface CalleeLike {
+    function swap(address src, address dst, uint256 amt, uint256 minOut, address to, bytes calldata data) external;
+}
+
+contract Swapper {
+    mapping (address => uint256) public wards;
+    mapping (address => mapping (address => uint256)) public hops; // [seconds]  hops[src][dst] is the swap cooldown when swapping `src` to `dst`.
+    mapping (address => mapping (address => uint256)) public zzz;  // [seconds]   zzz[src][dst] is the timestamp of the last swap from `src` to `dst`.
+    mapping (address => mapping (address => uint256)) public caps; // [weis]     caps[src][dst] is the maximum amount that can be swapped each hop when swapping `src` to `dst`.
+
+
+    address   public immutable buffer;                // Contract from which the GEM to sell is pulled and to which the bought GEM is pushed
+    RolesLike public immutable roles;                 // Contract managing access control for this Depositor
+    bytes32   public immutable ilk;
+
+    event Rely (address indexed usr);
+    event Deny (address indexed usr);
+    event File (bytes32 indexed what, address indexed src, address indexed dst, uint256 data);
+    event Swap (address indexed sender, address indexed src, address indexed dst, uint256 amt, uint256 out);
```

Comment:
```suggestion
    event Rely(address indexed usr);
    event Deny(address indexed usr);
    event File(bytes32 indexed what, address indexed src, address indexed dst, uint256 data);
    event Swap(address indexed sender, address indexed src, address indexed dst, uint256 amt, uint256 out);
```

Reply:
Done in https://github.com/makerdao/dss-allocator/pull/15